src/sentry/integrations/pagerduty/integration.py

```diff
@@ -69,12 +69,25 @@
 
 
 class PagerDutyIntegration(IntegrationInstallation):
-    def get_client(self, integration_key):
+    def get_keyring_client(self, keyid: str) -> PagerDutyProxyClient:
+        org_integration = self.org_integration
+        assert org_integration, "Cannot get client without an organization integration"
+
+        integration_key = None
+        for pds in org_integration.config.get("pagerduty_services", []):
+            if str(pds["id"]) == str(keyid):
```

Comment:
Curious why we need `str(keyid)` here as opposed to just `keyid`?

Reply:
I'm just being cautious as pagerduty services should have string ids all the time, but we have some tests for spike protection notifications where orm records with int values are used.